### ISSUE TITLE
Function to disable weapons "pushing" vehicles

### ISSUE THREAD

**Issue by Einheit-101** [multitheftauto/mtasa-blue#1346]
**Description**
(Moved from https://bugs.mtasa.com/view.php?id=9457)
Whenever you shoot a vehicle it moves slightly. Boats move a lot more.
I need a function to disable this, because you can move a giant Battleship with a minigun...

An additional weapon flag "push_vehicle" true/false would do the trick, which disables all velocity changes the weapon does to vehicles.

changing mass or turnMass does not fix it.

**How to reproduce**
-Spawn a Jetmax
-stand close to the Jetmax and fire with a minigun, it will float/rotate a lot.

**Additional context**
From mantis-9457.

**Comment by ThatG-I** [comment]
> I am currently working around this by getting the rudder state of my Battleship and compare it with the turn velocity of it, if the  
> ship is turning harder than it SHOULD with its current rudder rotation, the syncer of that ship limits the turn velocity to a value 
> that is in an acceptable range.

@Einheit-101 I see that you made a comment in **bugs.mtasa.com** and it's been a while now, but, by any chance did you find a better solution then the one you mentioned? It's been such a pain for me to deal with this problem.

**Comment by Einheit-101** [comment]
No, this is still how that massive movement is prevented. But iirc you can put damage of all weapons to 0 and they wont cause any pushing anymore

**Comment by ThatG-I** [comment]
Thank you, you might have just given me an idea. Store the damage amount in setElementData on that weapon, when it hit's just  getElementData and remove that damage amount from that vehicle or whatever.

**Comment by Pirulax** [comment]
Can't you change the physical mass of an object?

**Comment by PlatinMTA** [comment]
> Can't you change the physical mass of an object?

It was already pointed out that this does not work, unless you mean something else.

**Comment by ThatG-I** [comment]
> Can't you change the physical mass of an object?

A object is not really the problem it's mainly vehicles. Even if we was to change the mass it would effect the handling of the vehicle and that's not somthing that would be ideal for a good solution.

If their was a lot of progress in regards to https://github.com/multitheftauto/mtasa-blue/projects/14 then this problem would be solved, but in the mean time I think that being able to disable the bullet impact force on vehicles or in general is a decent solution.

**Comment by ThatG-I** [comment]
> No, this is still how that massive movement is prevented. But iirc you can put damage of all weapons to 0 and they wont cause any pushing anymore

I've tried that suggestion today and I cant seem to get that to work. Setting the weapon damage with setWeaponProperty func has no effect.
